### PR TITLE
Port to pixi and QuantCo Oss copier template

### PR DESCRIPTION
**Pull request by thomasmarwitz**
Checklist
* [x] Added a `docs/changes.rst` entry

### CODE REVIEW

**Comment by xhochy** [comment]
Please drop 3.8, we don't want to keep it as baggage.

**Comment by thomasmarwitz** [comment]
Regarding version bump: Is this minor or patch?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        run: pixi run -e build build-wheel
+      - name: Check package
+        run: pixi run -e build check-wheel
+      - name: Upload package
+        uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4.4.0
+        with:
+          name: artifact
+          path: dist/*
+
+  release:
```
Comment (thomasmarwitz) [inline .github/workflows/build.yml:31]: I have no way of testing this but it basically just the copier template code

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      - uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: artifact
+          path: dist
+      - name: Publish package on TestPyPi
+        uses: pypa/gh-action-pypi-publish@f7600683efdcb7656dec5b29656edb7bc586e597 # v1.10.3
+        with:
+          repository-url: https://test.pypi.org/legacy/
+          user: __token__
+          password: ${{ secrets.pypi_password }}
```
Comment (thomasmarwitz) [inline .github/workflows/build.yml:49]: Except the authentication I added based on the previous ci.yml workflow

**Comment by thomasmarwitz** [comment]
Lots of files were reformatted due to more pre-commit hooks. Also pyupgrade yielded some changes

File: .github/workflows/build.yml (hunk 2 above)
Comment (xhochy) [reply .github/workflows/build.yml:49]: We should change this to OIDC instead of using a password.

I can do the testing once we have merge.

**Comment by xhochy** [comment]
Is this ready for a review?

**Comment by thomasmarwitz** [comment]
> Is this ready for a review?
Yes

**Comment by thomasmarwitz** [comment]
@xhochy I added a changelog where I incremented the patch version.

**Review by xhochy — CHANGES_REQUESTED** [review]
One comment, otherwise this is LGTM

**Review by xhochy — APPROVED** [review]

**Pull request merged**